packages/medusa/src/utils/middlewares/authenticate-middleware.ts

```diff
@@ -20,13 +15,18 @@ const API_KEY_AUTH = "api-key"
 
 type AuthType = typeof SESSION_AUTH | typeof BEARER_AUTH | typeof API_KEY_AUTH
 
+const ADMIN_SCOPE = "admin"
+const STORE_SCOPE = "store"
+const ALL_SCOPE = "*"
+
+type Scope = typeof ADMIN_SCOPE | typeof STORE_SCOPE | typeof ALL_SCOPE
+
 type MedusaSession = {
-  auth_user: AuthUserDTO
-  scope: string
+  auth_context: AuthContext
 }
 
 export const authenticate = (
-  authScope: string | RegExp,
+  authScope: Scope | Scope[],
```

Comment:
todo: this should allow anything. If I create my own auth scope - e.g vendor.

Reply:
@srindom I'll bring this up tomorrow as well, I have a few more improvements I want to suggest, but I want to get everything working first.